### PR TITLE
RFC initial design

### PR DESCRIPTION
**Pull request by applejag**
Honest attemt at RFCs. This PR contains docs on how we think we should write RFCs, how we value them, and how we process them.

And not least, the template.

Please comment on the process, the template, or the docs themselves. Still, all of this can be changed after this has been merged, of course.
**The repo name is also up for discussion**. Having it as-is does mean it will collide with rust-lang/rfcs, emberjs/rfcs, and so on. Do we keep it or maybe change to have `wharf-` prefix on it as we have on the other repos? (https://github.com/iver-wharf/wharf-rfcs, leading to https://iver-wharf.github.io/wharf-rfcs)

I've changed GitHub pages to point to this branch, so you can see it rendered here:

- Index page: <https://iver-wharf.github.io/rfcs/>
- Template: <https://iver-wharf.github.io/rfcs/0000-template> (it's unlisted)

If this is merged, I'll also create the new labels `rfc` & `postponed`

### CODE REVIEW

File: 0000-template.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+<!--
+   Try to fill out the following sections. If nothing comes to mind for a
+   section, then literally write "Nothing comes to mind".
+
+   You are welcome to add more sections if you so need to.
+-->
+
+## Motivation
+
+Why do we need this? What's the problem you try to solve?
```
Comment (Pikabanga) [inline 0000-template.md]:
```suggestion
Why do we need this? What's the problem you are trying to solve?
```

File: 0000-template.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+func (mt MyType) String() string {
+    return fmt.Sprintf("%q %d", mt.text, mt.number)
+}
+```
+
+## Compatability
+
+Bring up compatability issues and things to regard. How will this interfere
+with existing components (providers, database, frontend)? Does this break
+backward compatability?
```
Comment (Pikabanga) [inline 0000-template.md]:
```suggestion
## Compatibility

Bring up compatibility issues and other things to regard. How will this interfere
with existing components (providers, database, frontend)? Does this break
backward compatibility?
```

File: README.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
+# Wharf RFCs (Request For Comments)
+
+For any substantial change that needs to be thought through in Wharf, we try to
+use RFCs for them instead of internally trying to talk it out in meetings.
+
+Not every descision needs an RFC, nor is any RFC excessive. Altough this is the
```
Comment (Pikabanga) [inline README.md]:
```suggestion
Not every decision needs an RFC, nor is any RFC excessive. This is the
```

File: README.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+start of our journey with RFCs, so we're trying to limit them to changes with
+major impact.
+
+Read more about the Wharf RFC process over at the GitHub pages:
+<https://iver-wharf.github.io/rfcs/>
+
+## Disclaimer
+
+The documents and markdown files in this repository are documentation of our
+descision making. Only to have a place to look back at, similar to meeting
```
Comment (Pikabanga) [inline README.md]:
```suggestion
decision making. Only to have a place to look back at, similar to meeting
```

File: README.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+notes or recordings, so we can see our previous descisions.
```
Comment (Pikabanga) [inline README.md]:
```suggestion
notes or recordings, so we can see our previous decisions.
```

File: README.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+**It is not** documentation of how things are meant to be implemented. Do not
+take this repository as the source of truth of how Wharf works. Please check
+out our [Wharf documentation](https://iver-wharf.github.io/) or the
+[source code](https://github.com/iver-wharf/) itself to gain the actual source
+of truth.
+
+## Running locally
+
+This repository is hosted using GitHub Pages, but you can also host it locally
+if you so want.
```
Comment (Pikabanga) [inline README.md]:
```suggestion
if you want.
```

File: _guides/rfc-life-cycle.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+2. **RFC goes into review.** Again, anyone may participate here. But the
+   developers/[collaborators](https://docs.github.com/en/github/getting-started-with-github/github-glossary#collaborator)
+   of Wharf have the final saying in any descision and discussion, especially
+   in *"endless discussions"*.
+
+   Our goal is that a newly created RFC is reviewed around 1-2 weeks after
+   creation.
+
+3. **A descision is made.** Will the RFC be rejected, published, or postponed?
```
Comment (Pikabanga) [inline _guides/rfc-life-cycle.md]:
```suggestion
3. **A decision is made.** Will the RFC be rejected, published, or postponed?
```

File: _guides/what-needs-an-rfc.md [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+
+Take it with a grain of salt. None of these rules are absolute, and there are
+lots of gray areas here.
+
+## Definition
+
+If your change you have in mind meets any of these critera, then you should
+*probably* create an RFC for it:
+
+1. Brings backward incomptabile changes to the backend or the way you host
```
Comment (Pikabanga) [inline _guides/what-needs-an-rfc.md]:
```suggestion
1. Brings backward incompatible changes to the backend or the way you host
```

File: index.md [hunk 9]
```diff
@@ -0,0 +1,31 @@
+---
+layout: default
+---
+
+# Wharf RFCs (Request For Comments)
+
+For any substantial change that needs to be thought through in Wharf, we try to
+use RFCs for them instead of internally trying to talk it out in meetings.
+
+Not every descision needs an RFC, nor is any RFC excessive. Altough this is the
```
Comment (Pikabanga) [inline index.md]:
```suggestion
Not every decision needs an RFC, nor is any RFC excessive. This is the
```

File: index.md [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+## Getting started
+
+Please visit the [RFC life cycle](./_guides/rfc-life-cycle.md) and
+[Writing RFCs](./_guides/writing-rfcs.md) pages to learn about how you can
+start proposing RFCs yourself.
+
+## Disclaimer
+
+The documents and markdown files in this repository are documentation of our
+descision making. Only to have a place to look back at, similar to meeting
```
Comment (Pikabanga) [inline index.md]:
```suggestion
decision making. Only to have a place to look back at, similar to meeting
```

File: index.md [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+notes or recordings, so we can see our previous descisions.
```
Comment (Pikabanga) [inline index.md]:
```suggestion
notes or recordings, so we can see our previous decisions.
```

**Review by Pikabanga — CHANGES_REQUESTED** [review]

**Comment by applejag** [comment]
If I got a nickel for every typo I made here...

Let's see here, 4 ~~compatability~~ compatibility, 7 ~~descision~~ decision... 0.55$! That's a FORTUNE!

**Review by Pikabanga — APPROVED** [review]
Great work!

I don't have any specific remarks. I think this is one of those things where we need to try it out and make adjustments over time.

**Review by iverestefans — APPROVED** [review]

**Pull request merged**